src/Day1.hs

```diff
@@ -113,6 +118,15 @@ clearTileMap c = fromList
 colorFromRoomIndex :: RoomIndex -> Color
 colorFromRoomIndex (RoomIndex idx) = colors !! (idx `mod` len)
   where
-    colors = colorWheel2
+    colors = rainbow
     len = length colors
+    
+  -- Tried to use TupleSections pragma but it wouldn't work
+wallTileMap :: [(Int, Int)] -> Map (Int, Int) Tile
+wallTileMap locs = fromList $ map (swap . (,) (Tile Nothing Nothing (Just bk2))) locs 
 
+doorTileMap :: [Int] -> [(Int, Int)] -> Map (Int, Int) Tile
+doorTileMap ids locs = fromList $ zipWith makeDoorTile ids locs
+
+makeDoorTile :: Int -> (Int, Int) -> ((Int, Int), Tile)
```

Comment:
move this to a where clause under `doorTileMap`

Reply:
Okay good idea.